docs(policy): add notes on DNS/L7 policies & Cilium agent availability (#20289)
cilium-team/cilium: Documentation/policy/language.rst

a single egress rule. See :ref:`DNS Obtaining Data` for information on
collecting this IP data.

.. note:: The DNS Proxy is provided in each Cilium agent.
   As a result, DNS requests targeted by policies depend on the availability
   of the Cilium agent pod.

>    This includes DNS visibility policies and pod annotations for policy.

Reviewer: Sorry, I didn't proof read my suggestion before I sent it. This is what I meant:
```suggestion
   This includes DNS visibility policies and pod annotations for visibility.
```

Author: Yes, I fixed it already :-)

.. note:: Layer 7 rules are not currently supported in `HostPolicies`, i.e.,
          policies that use :ref:`NodeSelector`.

.. note:: Layer 7 policies --and pod annotations-- result in traffic being

>    proxied through an Envoy container provided in each Cilium agent. As

Reviewer: ```suggestion
   proxied through an Envoy instance provided in each Cilium agent pod. As
```

Reviewer: ^^ Currently Envoy is just a process in the Cilium pod, but this could plausibly change at some point. Using `instance` here would cover both, so it won't be wrong if we forget to update this line later on :)

>    proxified through an Envoy instance provided in each Cilium agent pod.

Reviewer: Proxied would be the usual term for this. I saw before that the dictionary didn't include that term, but it is a more natural one than proxified.
```suggestion
   proxied through an Envoy instance provided in each Cilium agent pod.
```

Author: OK, I'll change that!

   a result, L7 traffic targeted by policies --including HTTP visibility
   policies-- or visibility pod annotations depend on the availability of the

>    Cilium agent.

Reviewer: It might be a bit easier to read these sentences if it is structured a bit simpler, something like:

> As a result, L7 traffic targeted by policies depend on the availability of the Cilium agent pod. This includes HTTP visibility policies and pod annotations for policy.

Arguably the last one could also be a link to the relevant section so it's clear what this statement refers to.

Author: Do you mean a link to pod annotations?

Reviewer: yeah this one: https://docs.cilium.io/en/stable/policy/visibility/#proxy-visibility

Author: Is the new commit correct in this regard?
